src/pessimisticPatternMatcherUtils.cc

```diff
@@ -284,18 +263,22 @@ ShiftRotMatrixResult create_shift_rot_matrix(double cos_rot_sq,
 
 bool intermediate_verify_comparison(std::vector<Eigen::Vector3d> const& src_pattern,
                                     std::vector<Eigen::Vector3d> const& ref_pattern,
-                                    Eigen::Matrix<double, 3, 3> const& shift_rot_matrix, double max_dist_rad,
-                                    int n_match) {
+                                    Eigen::Matrix3d const& shift_rot_matrix, double max_dist_rad) {
     double max_dist_sq = max_dist_rad * max_dist_rad;
-
     bool passed = true;
-    for (int idx = 0; idx < n_match; idx++) {
-        Eigen::Vector3d rot_src_vect = shift_rot_matrix * src_pattern[idx];
-        Eigen::Vector3d diff_vect = rot_src_vect - ref_pattern[idx];
+    auto iSrc = src_pattern.begin();
+    auto iRef = ref_pattern.begin();
+    while (iSrc != src_pattern.end() || iRef != ref_pattern.end()) {
+        Eigen::Vector3d rot_src_vect = shift_rot_matrix * *iSrc;
+        Eigen::Vector3d diff_vect = rot_src_vect - *iRef;
         if (max_dist_sq < diff_vect.dot(diff_vect)) {
             passed = false;
             break;
         }
+        if (iSrc != src_pattern.end()) {
+            iSrc++;
+            iRef++;
+        }
```

Comment:
I wouldn't mind using indexing here, but even with double iterators it's still more idiomatic to use a `for` loop:

```
for (auto iSrc = src_pattern.begin(), iRef = ref_pattern.begin(); iSrc != src_pattern.end() && iRef != ref_pattern.end(); iSrc++, iRef++)
```

(Note that the loop condition should be `&&`, not `||`; the latter will get you bad dereferencing if the two vectors don't have the same length.)